src/components/Commands.tsx

```diff
@@ -48,17 +45,130 @@ const Command = (props: any) => {
 }
 
 const Commands = () => {
+  const [loading, setLoading] = useState(false)
+
+  const [credentials] = useAsanaCredentials();
+  const [isAsanaTaskFormVisible, setIsAsanaTaskFormVisible] = useState(false);
+  const [asanaFormRef, setAsanaFormRef] = useState()
+  const { createTask: createAsanaTask } = useCreateAsanaTask();
+
+  const [isTaskFormVisible, setIsTaskFormVisible] = useState(false);
+  const [taskFormRef, setTaskFormRef] = useState()
+  const { createTask } = useCreateTask();
+  const client = useClient()
+
+  const isAsanaAuthorized = !!credentials.access_token
+  const commands = [{
+    name: 'Create Task',
+    icon: sunsamaLogo,
+    command: () => {
+      setIsTaskFormVisible(true)
+    }
+  }, {
+    name: isAsanaAuthorized ? 'Create Asana Task' : 'Connect to Asana',
+    icon: asanaLogo,
+    command: () => {
+      if (!isAsanaAuthorized) {
+        window.location.replace(client.app.asanaAuthorizeUrl())
+      } else {
+        setIsAsanaTaskFormVisible(true)
+      }
+    }
+  }]
+
+  const handleAsanaTaskCreateCancel = () => {
+    const { form } = asanaFormRef.props;
+    form.resetFields();
+    setIsAsanaTaskFormVisible(false)
+  }
```

Comment:
Today I learned. I didn't know you could access props from a child component like this. I would have just passed the `setIsAsanaTaskFormVisible` down to the child as a prop. I see the attraction in doing it this way instead.